indico/modules/rb_new/client/js/common/bookings/BookingDetailsModal.jsx

```diff
@@ -160,30 +170,184 @@ class BookingDetailsModal extends React.Component {
+                {showAccept && (
+                    <Button type="button"
+                            icon="check circle"
+                            color="green"
+                            size="small"
+                            onClick={() => this.changeState('approve')}
+                            loading={actionInProgress === 'approve' && bookingStateChangeInProgress}
+                            disabled={actionInProgress !== 'approve' && bookingStateChangeInProgress}
+                            content={<Translate>Accept booking</Translate>} />
+                )}
+            </Modal.Actions>
+        );
+    };
+
+    changeState = (action, data = {}) => {
+        const {booking: {id}, actions: {changeBookingState}} = this.props;
+        this.setState({actionInProgress: action}, () => {
+            changeBookingState(id, action, data).then(() => {
+                this.setState({actionInProgress: null});
+            });
+        });
+    };
+
+    renderBookingStatus = ({isPending, isAccepted, isCancelled, isRejected, rejectionReason}) => {
+        let color, status, icon, message;
+
+        if (isPending) {
+            icon = <Icon name="wait" />;
+            color = 'yellow';
+            status = Translate.string('Pending Confirmation');
+            message = Translate.string('The booking is subject to acceptance by the room owner');
+        } else if (isCancelled) {
+            icon = <Icon name="cancel" />;
+            color = 'grey';
+            status = Translate.string('Cancelled');
+            message = Translate.string('The booking was cancelled');
+        } else if (isRejected) {
+            icon = <Icon name="calendar minus" />;
+            color = 'red';
+            status = Translate.string('Rejected');
+            message = (
+                <>
+                    <Translate>
+                        The booking was rejected.
+                    </Translate>
+                    <br />
+                    <Translate>
+                        Reason: <Param name="rejectionReason" value={rejectionReason} wrapper={<strong />} />
+                    </Translate>
+                </>
+            );
+        } else if (isAccepted) {
+            icon = <Icon name="checkmark" />;
+            color = 'green';
+            status = Translate.string('Accepted');
+            message = Translate.string('The booking was accepted');
+        }
+
+        const label = (
+            <Label color={color}>
+                {icon}
+                {status}
+            </Label>
+        );
+
+        return (
+            <Popup trigger={label}
+                   content={message}
+                   position="bottom center" />
+        );
+    };
+
+    renderDeleteButton = () => {
+        const content = (
+            <>
+                <Message icon="exclamation triangle"
+                         header={Translate.string('Are you sure you want to delete this reservation?')} />
```

Comment:
What about using [Confirm](https://react.semantic-ui.com/addons/confirm/) for this? For something as dangerous as deletion (even if it's admin-only) I think I'd prefer something a bit more intrusive.